Source/WebCore/page/Page.cpp

```diff
@@ -4885,6 +4885,14 @@ void Page::performOpportunisticallyScheduledTasks(MonotonicTime deadline)
     if (m_opportunisticTaskScheduler->hasImminentlyScheduledWork())
         options.add(JSC::VM::SchedulerOptions::HasImminentlyScheduledWork);
     commonVM().performOpportunisticallyScheduledTasks(deadline, options);
+
+    LocalFrame* localMainFrame = dynamicDowncast<LocalFrame>(mainFrame());
+    if (!localMainFrame->document())
```

Comment:
I think there needs to be a check here if localMainFrame is non-null before accessing. I will amend this.